cpg-core/src/main/kotlin/de/fraunhofer/aisec/cpg/assumptions/Assumption.kt

```diff
@@ -0,0 +1,144 @@
+    // results have a decided state, assumptions are (not?) shown.
+    Confirmed, // User or Algorithm: Patterns(Paths) depending on the assumption are returned,
+    // results have a decided state, assumptions are shown.
+    Ignored, // User or Algorithm: Patterns(Paths) depending on the assumption are returned, results
+    // have a decided state, assumptions are not shown.
+}
+
+enum class AssumptionType {
+    SyntaxAmbiguityAssumption,
+    InferenceAssumption,
+    ClosedMacroAssumption,
+    UnsupportedLanguageProblem,
+    MissingCodeProblem,
+    AmbiguityAssumption,
+    ConceptAssumption,
+    ExhaustiveEnumerationAssumption,
+    CompletenessAssumption,
+    SoundnessAssumption,
+    ControlFlowAssumption,
+    CFIntegrityAssumption,
+    NoExceptionsAssumption,
+    CFAllOrNothingExecutesAssumption,
+    TrustedConfigAssumption,
+    DataFlowAssumption,
+    ExternalDataAssumption,
+    NetworkAvailableAssumption,
+    ResourceExistsAssumption,
+    ServiceReachableAssumption,
+    AtomicExecutionAssumption,
+    TrustBoundaryAssumption,
+    DataRangeAssumption,
+    TrustedInputAssumption,
+}
+
+/**
+ * @param assumptionType The type of assumption used to differentiate between assumptions and group
+ *   similar assumptions.
+ * @param node The node that cause the assumption to be necessary, even if the assumption has
+ *   validity for the entire program, the node serves as location to be reported to the user.
+ * @param scope The scope that the assumption has validity for, here the scope is a node, because
+ *   the assumption is valid for every node in its ast subtree.
+ * @param message The message describing the assumption that was taken.
+ */
+fun assume(assumptionType: AssumptionType, message: String, node: Node, scope: Node? = null) {
+    // This connects the assumption as an overlay node to the code graph
+    Assumption(assumptionType, message, getCurrentFileAndLine(), node, assumptionScope = scope)
+}
+
+/**
+ * @param assumptionType The type of assumption used to differentiate between assumptions and group
+ *   similar assumptions.
+ * @param edge The edge that cause the assumption to be necessary, even if the assumption has
+ *   validity for the entire program, the node that the edge starts at serves as location to be
+ *   reported to the user.
+ * @param scope The scope that the assumption has validity for, here the scope is a node, because
+ *   the assumption is valid for every node in its ast subtree.
+ * @param message The message describing the assumption that was taken.
+ */
+fun assume(assumptionType: AssumptionType, message: String, edge: Edge<*>?, scope: Node? = null) {
+    // This connects the assumption as an overlay node to the code graph
+    Assumption(
+        assumptionType,
+        message,
+        getCurrentFileAndLine(),
+        edge = edge,
+        assumptionScope = scope,
+    )
+}
+
+fun getCurrentFileAndLine(): String {
```

Comment:
missing doc. Why `stackTrace[4]`?! Will this crash if the stack trace is different to your expectations?

Reply:
This should never crash, especially after i made it private. To make it more transparent how this work, and not work off of a stack trace offset that should never change. I now look at the file names to be more resilient.